pkg/monitor/monitor/util.go

```diff
@@ -614,12 +654,18 @@ func getMonitorVolumes(config *core.ConfigMap, monitor *v1alpha1.TidbMonitor, tc
 	}
 	volumes = append(volumes, prometheusRules)
 	if tc.IsTLSClusterEnabled() {
+		var secretName string
+		if len(monitor.Spec.Clusters) > 1 {
```

Comment:
What about the heterogeneous clusters? https://github.com/pingcap/tidb-operator/pull/3297 updates the `spec.clusters` to include the TCs created by auto-scaling.